sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/HttpPostControllerActionShouldValidateInput.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Web.Mvc;
+
+namespace Tests.Diagnostics
+{
+    public class FooBarController1 : Controller
+    {
+        [HttpPost]
+        [ValidateInput(false)] // Noncompliant
+        public ActionResult Purchase1(string input)
+        {
+            return Foo(input);
+        }
+
+        [HttpPost]
+        [ValidateInput(true)] // Compliant - not false
+        public ActionResult Purchase2(string input)
+        {
+            return Foo(input);
+        }
+
+        [HttpPost]
+        [ValidateInput()] // Compliant - not false
+        public ActionResult Purchase3(string input)
+        {
+            return Foo(input);
+        }
+
+        [ValidateInput(false)] // Noncompliant - no HttpPost
+        public ActionResult Purchase4(string input)
+        {
+            return Foo(input);
+        }
+    }
+
+    public class FooBarController1
+    {
+        [HttpPost]
+        [ValidateInput(false)] // Noncompliant - class doesn't inherit from Controller
+        public ActionResult Purchase1(string input)
+        {
+            return Foo(input);
+        }
+    }
+}
```

Comment:
A couple of other tests cases (all of which are invalid code):
* [ValidateInput("foo")]  //  invalid code - not a boolean
* [ValidateInput(true, false]  //  invalid code - wrong number of attributes
* [Unrecognised]  //  invalid code - unrecognised attribute

Reply:
Done.

Reply:
Other test cases:
* method with _[ValidateInput(false)]_ but no _[HttpPost]_ attribute -> shouldn't raise
* fully qualified attribute names: just to check they are handled correctly, since the analyzer is doing _StartsWith_ check on _Name_